File: WikiCodeParser/Models/WikiRevision.cs
using System.Text.RegularExpressions;

namespace WikiCodeParser.Models
{
    public class WikiRevision
    {
        public static string CreateSlug(string text)
        {
            text = text.Replace(' ', '_');
            text = Regex.Replace(text, @"[^-$_.+!*\'""(),:;<>^{}|~0-9a-z[\]]", "", RegexOptions.IgnoreCase | RegexOptions.Singleline);
            return text;
        }
    }
}

File: javascript/src/Models/WikiRevision.ts

export class WikiRevision {
    public static CreateSlug(text : string) {
        text = text.replace(/ /ig, '_');
        text = text.replace(/[^-$_.+!*'"(),:;<>^{}|~0-9a-z[\]]/ig, '');
        return text;
    }
}
